### PR TITLE
Canvas: add alpha canvas panel and initial interfaces

### PR DESCRIPTION
**Pull request by ryantxu**
early WIP
text box
Screen Shot 2021-08-17 at 9 57 01 PM

### CODE REVIEW

**Review by lean257 — APPROVED** [review]

File: public/app/plugins/panel/canvas/runtime/scene.tsx [hunk 1]
```diff
@@ -0,0 +1,325 @@
+import React, { CSSProperties } from 'react';
```
Comment (mckn) [inline public/app/plugins/panel/canvas/runtime/scene.tsx:1]: nit: maybe worth splitting this up into multiple files?

File: public/app/types/events.ts [hunk 2]
```diff
@@ -201,3 +201,6 @@ export class AnnotationQueryFinished extends BusEventWithPayload<AnnotationQuery
 export class TimeRangeUpdatedEvent extends BusEventWithPayload<TimeRange> {
   static type = 'time-range-updated';
 }
+export class PanelEditFinishedEvent extends BusEventWithPayload<number> {
```
Comment (mckn) [inline public/app/types/events.ts:204]: When is this event published?

Comment (mckn) [reply public/app/types/events.ts:204]: Nvm, just saw it. Would it be better to name it something like `PanelEditExitedEvent` ?

Comment (ryantxu) [reply public/app/types/events.ts:204]: :+1: replaced

**Review by mckn — APPROVED** [review]
LGTM!

**Comment by ryantxu** [comment]
I'll merge... this is still super alpha, but good to have other PRs be more readable

**Pull request merged**